course/src/main/java/in/testpress/course/ui/BookmarksFragment.java

```diff
@@ -322,22 +328,47 @@ private void displayAttachmentContent() {
         } else {
             description.setVisibility(View.GONE);
         }
-        Button downloadButton = rootLayout.findViewById(R.id.download_attachment);
-        ViewUtils.setLeftDrawable(downloadButton.getContext(), downloadButton,
-                R.drawable.ic_file_download_18dp);
+    }
 
+    private void updateAttachmentButton(){
+        Button downloadButton = rootLayout.findViewById(R.id.download_attachment);
+        if (!content.getAttachment().getIsRenderable()) {
+            downloadButton.setText(R.string.testpress_download_attachment);
+            ViewUtils.setLeftDrawable(downloadButton.getContext(), downloadButton,
+                    R.drawable.ic_file_download_18dp);
+        }
         downloadButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                startActivity(new Intent(Intent.ACTION_VIEW,
-                        Uri.parse(attachment.getAttachmentUrl())));
+                if (content.getAttachment().getIsRenderable()){
```

Comment:
```suggestion
                if (content.getAttachment().getIsRenderable() == true){
```
It might come null also.